crates/templates/src/macros.rs

```diff
@@ -78,15 +78,18 @@ macro_rules! register_templates {
             /// # Errors
             ///
             /// Returns an error if any template fails to render with any of the sample.
-            pub(crate) fn all(templates: &Templates, now: chrono::DateTime<chrono::Utc>, rng: &mut impl rand::Rng) -> anyhow::Result<::std::collections::BTreeMap<(&'static str, SampleIdentifier), String>> {
+            pub(crate) fn all<R: Rng + Clone>(templates: &Templates, now: chrono::DateTime<chrono::Utc>, rng: &R) -> anyhow::Result<::std::collections::BTreeMap<(&'static str, SampleIdentifier), String>> {
```

Comment:
Instead of requiring a cloneable RNG here, we could seed a ChaChaRng out of the rng passed as parameter and clone it within the function. That would be a lot less suspicious :p

Reply:
I mean, either one is suspicious, but at least this one is a bit less prolific so sure.